exp/api/v1beta1/awsmanagedmachinepool_webhook.go

```diff
@@ -188,6 +214,16 @@ func (r *AWSManagedMachinePool) validateImmutable(old *AWSManagedMachinePool) fi
 	appendErrorIfMutated(old.Spec.AMIType, r.Spec.AMIType, "amiType")
 	appendErrorIfMutated(old.Spec.RemoteAccess, r.Spec.RemoteAccess, "remoteAccess")
 	appendErrorIfSetAndMutated(old.Spec.CapacityType, r.Spec.CapacityType, "capacityType")
+	if (old.Spec.AWSLaunchTemplate != nil && r.Spec.AWSLaunchTemplate == nil) ||
```

Comment:
Could we use the `appendErrorIfMutated` for this test?

Reply:
I was thinking that we should allow user to modify the launch template (e.g. update AMI ID), but should not allow user to change a node group from using launch template to not using launch template (and vice versa). That's why I didn't use `appendErrorIfMutated`. Let me know if that makes sense. Thanks!